.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,385 @@
+            run_id=$(echo "$details" | jq --arg repo "$repo" '".$repo"."run_id"')
+            echo "Discovered run_id '$run_id' for repo $repo"
+            all_details=$(echo "$details" | jq --arg repo "$repo" '".$repo"')
+            elif [[ -n "$run_id" ]]; then
+              filtered_details=$(echo "$filtered_details" | jq --arg all_details "$all_details" --arg repo "$repo" '. + {"$repo": $all_details}')
+            else
+              echo "Skipping (empty run_id)"
+            fi
+          done <<(echo "$details" | jq -rc 'keys[]')
+          echo "Filtered upstream details: $filtered_details"
+          echo "upstream-details=$(echo $filtered_details)" >> $GITHUB_OUTPUT
+  
+  set-version:
+    needs: ['set-metadata']
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.choose-version.outputs.version }}
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get previous git tag
+        id: previous-tag
+        run: |
+          tag=$(git for-each-ref --sort=-creatordate --count 5 --format="%(refname:short)" "refs/tags/" | grep -E "^[0-9]+\.[0-9]+\.[0-9]+" | head -n 1)
+          echo "Found tag: $tag"
+          echo "tag=$(echo tag)" >> $GITHUB_OUTPUT
+
+      - name: Get next semver from previous tag
+        id: next-semvers
+        uses: WyriHaximus/github-action-next-semvers@v1
+        with:
+          version: ${{ steps.previous-tag.outputs.tag }}
+
+      - name: Set PR version
+        if: needs.set-metadata.outputs.build-type == 'PR'
+        id: set-pr-version
+        run: |
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          PR_ITERATION=${{ github.run_attempt }}.${{ github.run_number }}
+          echo "version=$(echo ${{ steps.next-semvers.outputs.patch }}-alpha+PR.${PR_NUMBER}.${PR_ITERATION})" >> $GITHUB_OUTPUT
+
+      - name: Set Branch version
+        if: needs.set-metadata.outputs.build-type == ''
+        id: set-branch-version
+        run: |
+          BRANCH_ITERATION=${{ github.run_attempt }}.${{ github.run_number }}
+          echo "version=$(echo ${{ steps.next-semvers.outputs.patch }}-alpha+BR.${{ github.ref_name }}.${BRANCH_ITERATION})" >> $GITHUB_OUTPUT
+
+      - name: Make Version PEP-Compliant
+        if: steps.set-pr-version != '' || steps.set-branch-version != ''
+        id: pepify
+        run: |
+          if [[ -n "${{ steps.set-pr-version.outputs.version }}" ]]; then
+            echo "Found PR version to pepify..."
+            semver=${{ steps.set-pr-version.outputs.version }}
+          else
+            echo "Found branch version to pepify..."
+            semver=${{ steps.set-branch-version.outputs.version }}
+          fi
+          # If this is not a Python repo, do nothing
+          if ! [[ "${{ github.repository }}" =~ "-py" ]]; then
+            echo "Current repository is not a Python project, nothing to PEP-ify."
+            echo "version=$semver" >> $GITHUB_OUTPUT
+          fi
```

Comment:
I would move the non-python check first so that the log doesn't have those echo messages from the previous if/else block that don't apply. Also looks like it is missing an 'exit 0' statement.